logger.go

```diff
@@ -111,21 +120,22 @@ func (l *Logger) Log(p Priority, xs ...interface{}) {
 	addF := func(bf F) {
 		if event.Data == nil {
 			event.Data = bf
-		} else {
-			for k, v := range bf {
-				event.Data[k] = v
-			}
+			return
+		}
+		for k, v := range bf {
+			event.Data[k] = v
 		}
 	}
 
 	// Assemble the event
 	for _, b := range xs {
-		if bf, ok := b.(F); ok {
-			addF(bf)
-		} else if fer, ok := b.(Fer); ok {
-			addF(F(fer.F()))
-		} else {
-			bits = append(bits, b)
+		switch t := b.(type) {
```

Comment:
This is a really, really, odd and unintuitive construct, even though it is documented in the specification, it trips me up every time.